### PR TITLE
Use Github Workflows over CircleCi in Gnarails, setup in downstream apps

### LINKED ISSUE #18: Resolve rubocop warnings out of box
After running `rails new` with this and running `bundle exec rubocop`, the generated code will result in rubocop warnings based on our style guide. We should update this template to ensure that this starts with a clean rubocop result.

### LINKED ISSUE #50: Address rubocop violations
After introducing gnar-style #45, we should update the repo to adhere to the conventions from the style file.

### LINKED ISSUE #136: Brakeman Binstub Conflict
Currently, we place a script in `bin/brakeman`, which is used to run brakeman, particularly in CI.

However, if your project is generating binstubs, brakeman *will* place a binstub in `bin/brakeman`. These files will conflict, but should exist separately.

I think the solution here is as easy as renaming the templated `bin/brakeman`, similar to `bin/ci_pronto`.

### PR DESCRIPTION
**Pull request by alxjrvs**
Closes #18, #50 and #136.

This PR moves gnarails - and apps created from it - completely off of circleci and towards a glorious Github Actions-based future. There are a number of other changes that I will get into more detail in below. 

This change allows us to break out functionality previously left as steps in a single asynchronous job. Pronto, Brakeman, and Bundle-Audit are all broken out into their own actions, and can report independently. This allowed us to change the dependencies a bit and leave some gems for their specific implementation in the workflow action. 

Additionally, this separation allows us to share a great deal between gnarails and apps that it would generate, which makes them more resilient. A little dogfood never hurt anyone. 

There are a number of changes to associated files to try and bring this up to the functionality of circleci.  This involved creating a number of binstubs, editing Gnarly, and moving a lot of files around. This PR is less noisy than it seems! 

I have squashed down my terrifying journey so that none of you may walk it. 

Closes #18 - Rubocop warnings now properly resolved. 
Closes #50 - Verifying Rubocop is free of serious warnings in CI 
closes #136 - No more Brakeman binstub, no more Brakean Binstub warning.

### CODE REVIEW

File: .github/workflows/rails-test-app.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          mkdir rails-test-app/spec/factories
+          mkdir rails-test-app/spec/models
+          mkdir rails-test-app/spec/system
+          mkdir rails-test-app/spec/requests
+
+          cp test-app/app/controllers/job_postings_controller.rb rails-test-app/app/controllers/job_postings_controller.rb
+          cp test-app/app/models/job_posting.rb rails-test-app/app/models/job_posting.rb
+          cp test-app/app/models/comment.rb rails-test-app/app/models/comment.rb
+          cp test-app/app/views/job_postings/index.html.erb rails-test-app/app/views/job_postings/index.html.erb
+          yes | cp test-app/app/views/layouts/application.html.erb rails-test-app/app/views/layouts/application.html.erb
```
Comment (taylorkearns) [inline .github/workflows/rails-test-app.yml:39]: `yes`?

**Review by taylorkearns — COMMENTED** [review]
This all looks fine to me; would like to get @mikestone14 or @ngmaloney 's sign off on these changes before we go ahead.

Is using GH free?

**Comment by alxjrvs** [comment]
@taylorkearns My understanding is that, under our current plans, public repos are free. Private repos have a CI minutes/month limit. 

I should also mention that I *think* it might be in our bests interests to sunset `gnarails` in the long run, and it took me going through this (and gnar-cli) in the long-run. See this [follow-up pr](https://github.com/TheGnarCo/gnarails/pull/170) to gnarails and the [implementation](https://github.com/TheGnarCo/gnar-cli/pull/30) over in gnar-cli for more context.

**Comment by efertsch** [comment]
@alxjrvs great work on this. Looks like there is one rubocop warning that needs to be addressed. This seems worthwhile to me, given that CircleCI could boot us from the legacy plan at any time, but I agree with @taylorkearns that it makes sense to involve @mikestone14 & @ngmaloney of changes like these 😄

**Comment by alxjrvs** [comment]
@efertsch I'm personally in favor of letting that one rubocop error stand, it's a comment in the template that would be awkward to arrange otherwise. That said, thanks!

**Review by efertsch — APPROVED** [review]

File: .github/actions/test-rails/action.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: 'Test Rails - The Gnar Way'
+description: 'Setup and run the tests for your Rails app'
+runs:
+  using: "composite"
```
[the remark this answers is not in the document]
Comment (alxjrvs) [reply .github/actions/test-rails/action.yml:4]: [Composite Actions](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) are github actions that can be called by other workflows. This action (test-rails) is called by the `test-rails-app` workflow. 

Think of this action as the very narrow "minium actions required to run the rails app tests" and `test-rails-app` as doing the rest of the setup.

**Pull request merged**